zerver/webhooks/newrelic/view.py

```diff
@@ -1,169 +1,214 @@
 ) -> HttpResponse:
-    # Handle old format
-    # Once old is EOLed, delete if block and keep else block
-    if not payload.get("id").tame(check_none_or(check_string_fixed_length(36))):
-        info = {
-            "condition_name": payload.get("condition_name", "Unknown condition").tame(check_string),
-            "details": payload.get("details", "No details.").tame(check_string),
-            "incident_url": payload.get("incident_url", "https://alerts.newrelic.com").tame(
-                check_string
-            ),
-            "incident_acknowledge_url": payload.get(
-                "incident_acknowledge_url", "https://alerts.newrelic.com"
-            ).tame(check_string),
-            "status": payload.get("current_state", "None").tame(check_string),
-            "iso_timestamp": "",
-            "owner": payload.get("owner", "").tame(check_string),
-        }
-
-        unix_time = payload.get("timestamp").tame(
-            check_none_or(check_union([check_string, check_int]))
-        )
-        if unix_time is None:
-            raise JsonableError(_("The newrelic webhook requires timestamp in milliseconds"))
-
-        info["iso_timestamp"] = str(unix_milliseconds_to_timestamp(unix_time, "newrelic"))
-
-        # Add formatting to the owner field if owner is present
-        if info["owner"] != "":
-            info["owner"] = "by **{}** ".format(info["owner"])
-
-        # These are the three promised current_state values
-        if info["status"].lower() == "open":
-            content = OPEN_TEMPLATE.format(**info)
-        elif info["status"].lower() == "acknowledged":
-            content = DEFAULT_TEMPLATE.format(**info)
-        elif info["status"].lower() == "closed":
-            content = DEFAULT_TEMPLATE.format(**info)
-        else:
-            raise JsonableError(
-                _("The newrelic webhook requires current_state be in [open|acknowledged|closed]")
-            )
-
-        topic_info = {
-            "policy_name": payload.get("policy_name", "Unknown Policy").tame(check_string),
-            "incident_id": payload.get("incident_id", "Unknown ID").tame(
-                check_union([check_string, check_int])
-            ),
-        }
-        topic_name = TOPIC_TEMPLATE.format(**topic_info)
+    missing_fields = check_for_expected_fields(payload)
+    if missing_fields:
+        formatted_missing_fields = ", ".join([f"`{fields}`" for fields in missing_fields])
+        header = f":danger: An [incident]({DEFAULT_NEWRELIC_URL}) updated"
+        body_detail = f"**Warning**: Unable to use the default notification format because at least one expected field was missing from the incident payload. See [New Relic integration documentation](/integrations/doc/newrelic).\n\n**Missing fields**: {formatted_missing_fields}"
+        body = NOTIFICATION_TEMPLATE.format(header=header, body=body_detail).strip()
+        topic = "Incident alerts"
+        check_send_webhook_message(request, user_profile, topic, body)
```

Comment:
If we call `json_success` here for this and have an early return pattern, so that we can remove the `else` clause/indentation below.